### PR TITLE
Trigger password rotation

### PR DESCRIPTION
**Pull request by quaxsze**

### CODE REVIEW

File: udata/migrations/2020-10-06-add-password-rotation-needed.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'''
```
Comment (abulte) [inline udata/migrations/2020-10-06-add-password-rotation-needed.py:1]: needed w/ a default value to `False`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,9 @@
 - Using the fs_filename logic when uploading a new resource on the data catalog.[#2547](https://github.com/opendatateam/udata/pull/2547)
 - Remove old file when updating resources and community resources from API [#2548](https://github.com/opendatateam/udata/pull/2548)
 - Sortable.js upgrade to fix an issue in udata's editorial page when reordering featured datasets [#2550](https://github.com/opendatateam/udata/pull/2550)
+- Password rotation mechanism [#2551](https://github.com/opendatateam/udata/pull/2551):
+  - Date fields 'password_rotation_demand' and 'password_rotation_performed' added to user model.
```
Comment (abulte) [inline CHANGELOG.md:17]:
```suggestion
  - Datetime fields `password_rotation_demand` and `password_rotation_performed` added to user model.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,9 @@
...
+  - Override Flask-Security's lofin and reset password forms to implement the password rotation checks.
```
Comment (abulte) [inline CHANGELOG.md:18]:
```suggestion
  - Override Flask-Security's login and reset password forms to implement the password rotation checks.
```

File: udata/api/__init__.py [hunk 4]
```diff
@@ -106,6 +106,9 @@ def wrapper(*args, **kwargs):
             if current_user.deleted:
                 self.abort(401)
 
+            if current_user.password_rotation_demand:
```
Comment (abulte) [inline udata/api/__init__.py:109]: do we rely on user/password at the API level? Is it not only API key related?

File: udata/auth/__init__.py [hunk 5]
```diff
@@ -39,13 +39,18 @@ def __init__(self, *needs):
...
-    from .forms import ExtendedRegisterForm
+    from .forms import ExtendedRegisterForm, ExtendedLoginForm, ExtendedResetPasswordForm
     from .tasks import sendmail_proxy
     from .views import create_security_blueprint
     from .password_validation import password_validator
     from udata.models import datastore
-    state = security.init_app(app, datastore, register_blueprint=False,
-                              confirm_register_form=ExtendedRegisterForm, send_mail=sendmail_proxy)
+    state = security.init_app(app, datastore,
+                                register_blueprint=False,
```
Comment (abulte) [inline udata/auth/__init__.py:48]: strange indentation 🧐

File: udata/core/user/models.py [hunk 6]
```diff
@@ -65,6 +65,9 @@ class User(WithMetrics, UserMixin, db.Document):
     # when SECURITY_CONFIRMABLE is True
     confirmed_at = db.DateTimeField()
 
+    password_rotation_demand = db.DateTimeField()
```
Comment (abulte) [inline udata/core/user/models.py:68]:
```suggestion
    password_rotation_demanded = db.DateTimeField()
```

or 

```suggestion
    password_rotation_needed = db.DateTimeField()
```

Comment (abulte) [reply udata/core/user/models.py:68]: Ideally we should even add a `_at` suffix to indicate it's a date, but that might make the field names a bit long.

**Review by abulte — COMMENTED** [review]
Maybe add a quick doc "how to ask for a password rotation" somewhere

File: udata/api/__init__.py (hunk 4 above)
Comment (quaxsze) [reply udata/api/__init__.py]: Yes i might have been a bit zealous.

**Review by abulte — APPROVED** [review]

**Pull request merged**